go/api/base_client.go

```diff
@@ -0,0 +1,73 @@
+// Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
+
+package api
+
+// #cgo LDFLAGS: -L../target/release -lglide_rs
+// #include "../lib.h"
+//
+// void successCallback(uintptr_t channelPtr, char *message);
+// void failureCallback(uintptr_t channelPtr, char *errMessage, RequestErrorType errType);
+import "C"
+
+import (
+	"unsafe"
+
+	"github.com/aws/glide-for-redis/go/glide/protobuf"
+	"google.golang.org/protobuf/proto"
+)
+
+//export successCallback
+func successCallback(channelPtr C.uintptr_t, cResponse *C.char) {
+	// TODO: Implement when we implement the command logic
+}
+
+//export failureCallback
+func failureCallback(channelPtr C.uintptr_t, cErrorMessage *C.char, cErrorType C.RequestErrorType) {
+	// TODO: Implement when we implement the command logic
+}
+
+type clientConfiguration interface {
+	toProtobuf() *protobuf.ConnectionRequest
+}
+
+type baseClient struct {
+	coreClient unsafe.Pointer
+}
+
+func createClient(config clientConfiguration) (*baseClient, error) {
```

Comment:
Add a doc

Reply:
any function or variable that starts with a lower case letter is internal (ie private). The external (ie public) functions that call this are already documented (NewRedisClient and NewRedisClusterClient)